### PR TITLE
Fixes #761 - component.elId() does not work on the server

### LINKED ISSUE #761: component.elId() does not work on the server
The following component will render fine in the browser, but on the server the user will get an error (`TypeError: component.elId is not a function`):

```marko
class {}
<div>
  <input key="checkbox"/>
  <label for=component.elId('checkbox')/>
</div>
```

### PR DESCRIPTION
**Pull request by patrick-steele-idem**

### CODE REVIEW

File: test/autotests/render/component-elId/expected.html [hunk 1]
```diff
@@ -0,0 +1 @@
+Hello World!
```
Comment (austinkelleher) [inline test/autotests/render/component-elId/expected.html]: Assertion is incorrect causing the tests to fail.

File: src/components/ServerComponent.js [hunk 2]
```diff
@@ -52,6 +52,20 @@ class ServerComponent {
...
         return this.___state;
     }
+
+    elId(scopedId, index) {
+        var id = this.id;
+
+        var elId = scopedId != null ? id + "-" + scopedId : id;
+
+        if (index != null) {
+            elId += "[" + index + "]";
```
Comment (austinkelleher) [inline src/components/ServerComponent.js]: Maybe add a test where `index` is passed to?

File: test/autotests/render/component-getElId/expected.html [hunk 3]
```diff
@@ -0,0 +1 @@
+Hello World!
```
Comment (austinkelleher) [inline test/autotests/render/component-getElId/expected.html]: Assertion is incorrect causing the tests to fail.

**Comment by patrick-steele-idem** [comment]
Merged

**Pull request closed without merging**